File: backend/test/php/demo_php_vulnerable_project/db/orders_sql.php
<?php
// db/orders_sql.php

function search_orders($connection) {
    $userId = $_GET["user_id"];
    $status = $_GET["status"];
    $sortBy = $_GET["sort_by"];

    $selectedUser = $userId;
    $selectedStatus = $status;
    $orderColumn = $sortBy;

    $tableName = "orders";
    $baseQuery = "SELECT * FROM " . $tableName;

    $whereClause =
        " WHERE user_id = " . $selectedUser .
        " AND status = '" . $selectedStatus . "'";

    $orderClause = " ORDER BY " . $orderColumn;

    $query = $baseQuery . $whereClause . $orderClause;

    mysqli_query($connection, $query);
}


function update_order_status($connection) {
    $orderId = $_POST["order_id"];
    $newStatus = $_POST["status"];

    $query = "UPDATE orders SET status = '" . $newStatus . "' WHERE id = " . $orderId;

    mysqli_query($connection, $query);
}


function delete_order($connection) {
    $orderId = $_GET["order_id"];

    $query = "DELETE FROM orders WHERE id = " . $orderId;

    mysqli_query($connection, $query);
}
?>

File: backend/test/python/demo_python_vulnerable_project/db/orders_sql.py
# db/orders_sql.py

def search_orders(request, cursor):
    user_id = request.GET["user_id"]
    status = request.GET["status"]
    sort_by = request.GET["sort_by"]

    selected_user = user_id
    selected_status = status
    order_column = sort_by

    table_name = "orders"
    base_query = "SELECT * FROM " + table_name

    where_clause = (
        " WHERE user_id = " + selected_user +
        " AND status = '" + selected_status + "'"
    )

    order_clause = " ORDER BY " + order_column

    query = base_query + where_clause + order_clause

    cursor.execute(query)


def update_order_status(request, cursor):
    order_id = request.POST["order_id"]
    new_status = request.POST["status"]

    query = "UPDATE orders SET status = '" + new_status + "' WHERE id = " + order_id

    cursor.execute(query)


def delete_order(request, cursor):
    order_id = request.GET["order_id"]

    query = "DELETE FROM orders WHERE id = " + order_id

    cursor.execute(query)